Failed to run `make go-test` failed locally

```
=== NAME  TestReadAffinity_GetReadAffinityMapOptions
    nodeserver_test.go:249: failed to create directory /etc/ceph-csi-config: mkdir /etc/ceph-csi-config: permission denied
    nodeserver_test.go:253: failed to write /etc/ceph-csi-config/config.json file content: open /etc/ceph-csi-config/config.json: no such file or directory
=== RUN   TestReadAffinity_GetReadAffinityMapOptions/Enabled_in_cluster-1_and_Enabled_in_CLI
=== CONT  TestIsMigrationVolID/wrong_volume_ID
=== CONT  TestIsMigrationVolID/wrong_mons_prefixed_volume_ID
=== CONT  TestIsMigrationVolID/Wrong_volume_ID
--- PASS: TestIsMigrationVolID (0.00s)
    --- PASS: TestIsMigrationVolID/correct_volume_ID (0.00s)
    --- PASS: TestIsMigrationVolID/wrong_image_prefixed_volume_ID (0.00s)
    --- PASS: TestIsMigrationVolID/wrong_volume_ID (0.00s)
    --- PASS: TestIsMigrationVolID/wrong_mons_prefixed_volume_ID (0.00s)
    --- PASS: TestIsMigrationVolID/Wrong_volume_ID (0.00s)
--- PASS: TestNodeServer_appendReadAffinityMapOptions (0.00s)
    --- PASS: TestNodeServer_appendReadAffinityMapOptions/both_empty_mapOptions_and_crushLocationMap (0.00s)
    --- PASS: TestNodeServer_appendReadAffinityMapOptions/filled_mapOptions,_empty_readAffinityMapOptions_&_non-default_mounter (0.00s)
    --- PASS: TestNodeServer_appendReadAffinityMapOptions/filled_mapOptions,_filled_crushLocationMap_&_non-default_mounter (0.00s)
    --- PASS: TestNodeServer_appendReadAffinityMapOptions/filled_mapOptions,_empty_readAffinityMapOptions_&_default_mounter (0.00s)
    --- PASS: TestNodeServer_appendReadAffinityMapOptions/empty_mapOptions,_filled_crushLocationMap_&_default_mounter (0.00s)
    --- PASS: TestNodeServer_appendReadAffinityMapOptions/filled_mapOptions,_filled_crushLocationMap_&_default_mounter (0.00s)
    --- PASS: TestNodeServer_appendReadAffinityMapOptions/empty_mapOptions,_filled_crushLocationMap_&_non-default_mounter (0.00s)
=== CONT  TestParseMigrationVolID/correct_volume_ID
=== CONT  TestParseMigrationVolID/volume_ID_without_'mon-'_prefix_string
=== CONT  TestParseMigrationVolID/volume_ID_with_unallowed_image_name
=== CONT  TestParseMigrationVolID/volume_ID_with_unallowed_migration_version_string
=== CONT  TestParseMigrationVolID/correct_volume_ID_with_'_'_pool_name
=== CONT  TestParseMigrationVolID/correct_volume_ID_with_more_than_one_mon
=== CONT  TestParseMigrationVolID/correct_volume_ID_with_single_mon
=== CONT  TestParseMigrationVolID/volume_ID_without_pool
=== CONT  TestParseMigrationVolID/volume_ID_without_image
=== CONT  TestParseMigrationVolID/volume_ID_without_mons
--- PASS: TestParseMigrationVolID (0.00s)
    --- PASS: TestParseMigrationVolID/correct_volume_ID (0.00s)
    --- PASS: TestParseMigrationVolID/volume_ID_without_'mon-'_prefix_string (0.00s)
    --- PASS: TestParseMigrationVolID/volume_ID_with_unallowed_image_name (0.00s)
    --- PASS: TestParseMigrationVolID/volume_ID_with_unallowed_migration_version_string (0.00s)
    --- PASS: TestParseMigrationVolID/correct_volume_ID_with_'_'_pool_name (0.00s)
    --- PASS: TestParseMigrationVolID/correct_volume_ID_with_more_than_one_mon (0.00s)
    --- PASS: TestParseMigrationVolID/volume_ID_without_pool (0.00s)
    --- PASS: TestParseMigrationVolID/volume_ID_without_image (0.00s)
    --- PASS: TestParseMigrationVolID/volume_ID_without_mons (0.00s)
    --- PASS: TestParseMigrationVolID/correct_volume_ID_with_single_mon (0.00s)
=== PAUSE TestReadAffinity_GetReadAffinityMapOptions/Enabled_in_cluster-1_and_Enabled_in_CLI
E0515 08:45:59.837777 2433752 rbd_util.go:361] krbd feature "journaling" not supported
--- PASS: TestIsKrbdFeatureSupported (0.00s)
    --- PASS: TestIsKrbdFeatureSupported/supported_feature (0.00s)
    --- PASS: TestIsKrbdFeatureSupported/not_supported_feature (0.00s)
=== RUN   TestReadAffinity_GetReadAffinityMapOptions/Disabled_in_cluster-2_and_Enabled_in_CLI
E0515 08:45:59.838103 2433752 nodeserver.go:123] failed to parse value of "myOption": ""
E0515 08:45:59.841158 2433752 nodeserver.go:123] failed to parse value of "myOption": "non-boolean"
=== PAUSE TestReadAffinity_GetReadAffinityMapOptions/Disabled_in_cluster-2_and_Enabled_in_CLI
--- PASS: TestParseBoolOption (0.00s)
=== RUN   TestReadAffinity_GetReadAffinityMapOptions/Enabled_in_cluster-3_with_empty_crush_labels_and_Enabled_in_CLI
=== PAUSE TestReadAffinity_GetReadAffinityMapOptions/Enabled_in_cluster-3_with_empty_crush_labels_and_Enabled_in_CLI
=== RUN   TestReadAffinity_GetReadAffinityMapOptions/Enabled_in_cluster-3_with_empty_crush_labels_and_Disabled_in_CLI
=== PAUSE TestReadAffinity_GetReadAffinityMapOptions/Enabled_in_cluster-3_with_empty_crush_labels_and_Disabled_in_CLI
=== RUN   TestReadAffinity_GetReadAffinityMapOptions/Absent_in_cluster-4_and_Enabled_in_CLI
=== PAUSE TestReadAffinity_GetReadAffinityMapOptions/Absent_in_cluster-4_and_Enabled_in_CLI
=== CONT  TestReadAffinity_GetReadAffinityMapOptions/Enabled_in_cluster-1_and_Enabled_in_CLI
=== CONT  TestReadAffinity_GetReadAffinityMapOptions/Enabled_in_cluster-3_with_empty_crush_labels_and_Disabled_in_CLI
=== CONT  TestReadAffinity_GetReadAffinityMapOptions/Enabled_in_cluster-3_with_empty_crush_labels_and_Enabled_in_CLI
=== CONT  TestReadAffinity_GetReadAffinityMapOptions/Disabled_in_cluster-2_and_Enabled_in_CLI
=== CONT  TestReadAffinity_GetReadAffinityMapOptions/Absent_in_cluster-4_and_Enabled_in_CLI
=== NAME  TestReadAffinity_GetReadAffinityMapOptions/Enabled_in_cluster-3_with_empty_crush_labels_and_Enabled_in_CLI
    nodeserver_test.go:310: 
        	Error Trace:	/home/mrajanna/workspace/src/github.com/ceph/ceph-csi/internal/rbd/nodeserver_test.go:310
        	Error:      	error fetching configuration for cluster ID "cluster-3": open /etc/ceph-csi-config/config.json: no such file or directory
        	Test:       	TestReadAffinity_GetReadAffinityMapOptions/Enabled_in_cluster-3_with_empty_crush_labels_and_Enabled_in_CLI
=== NAME  TestReadAffinity_GetReadAffinityMapOptions/Enabled_in_cluster-3_with_empty_crush_labels_and_Disabled_in_CLI
    nodeserver_test.go:310: 
        	Error Trace:	/home/mrajanna/workspace/src/github.com/ceph/ceph-csi/internal/rbd/nodeserver_test.go:310
        	Error:      	error fetching configuration for cluster ID "cluster-3": open /etc/ceph-csi-config/config.json: no such file or directory
        	Test:       	TestReadAffinity_GetReadAffinityMapOptions/Enabled_in_cluster-3_with_empty_crush_labels_and_Disabled_in_CLI
=== NAME  TestReadAffinity_GetReadAffinityMapOptions/Disabled_in_cluster-2_and_Enabled_in_CLI
    nodeserver_test.go:310: 
        	Error Trace:	/home/mrajanna/workspace/src/github.com/ceph/ceph-csi/internal/rbd/nodeserver_test.go:310
        	Error:      	error fetching configuration for cluster ID "cluster-2": open /etc/ceph-csi-config/config.json: no such file or directory
        	Test:       	TestReadAffinity_GetReadAffinityMapOptions/Disabled_in_cluster-2_and_Enabled_in_CLI
=== NAME  TestReadAffinity_GetReadAffinityMapOptions/Enabled_in_cluster-1_and_Enabled_in_CLI
    nodeserver_test.go:310: 
        	Error Trace:	/home/mrajanna/workspace/src/github.com/ceph/ceph-csi/internal/rbd/nodeserver_test.go:310
        	Error:      	error fetching configuration for cluster ID "cluster-1": open /etc/ceph-csi-config/config.json: no such file or directory
        	Test:       	TestReadAffinity_GetReadAffinityMapOptions/Enabled_in_cluster-1_and_Enabled_in_CLI
=== NAME  TestReadAffinity_GetReadAffinityMapOptions/Absent_in_cluster-4_and_Enabled_in_CLI
    nodeserver_test.go:310: 
        	Error Trace:	/home/mrajanna/workspace/src/github.com/ceph/ceph-csi/internal/rbd/nodeserver_test.go:310
        	Error:      	error fetching configuration for cluster ID "cluster-4": open /etc/ceph-csi-config/config.json: no such file or directory
        	Test:       	TestReadAffinity_GetReadAffinityMapOptions/Absent_in_cluster-4_and_Enabled_in_CLI
--- FAIL: TestReadAffinity_GetReadAffinityMapOptions (0.00s)
    --- FAIL: TestReadAffinity_GetReadAffinityMapOptions/Enabled_in_cluster-3_with_empty_crush_labels_and_Enabled_in_CLI (0.01s)
    --- FAIL: TestReadAffinity_GetReadAffinityMapOptions/Enabled_in_cluster-3_with_empty_crush_labels_and_Disabled_in_CLI (0.02s)
    --- FAIL: TestReadAffinity_GetReadAffinityMapOptions/Disabled_in_cluster-2_and_Enabled_in_CLI (0.02s)
    --- FAIL: TestReadAffinity_GetReadAffinityMapOptions/Enabled_in_cluster-1_and_Enabled_in_CLI (0.02s)
    --- FAIL: TestReadAffinity_GetReadAffinityMapOptions/Absent_in_cluster-4_and_Enabled_in_CLI (0.02s)
FAIL
FAIL	github.com/ceph/ceph-csi/internal/rbd	0.097s
```

Test should not fail when we run on a local machine, the problem is with the config.json file path, we need to create it with test folder with `tmpConfPath := t.TempDir() + "/ceph-csi.json"` like other tests

--- a/internal/rbd/nodeserver_test.go
+++ b/internal/rbd/nodeserver_test.go
@@ -243,11 +243,8 @@ func TestReadAffinity_GetReadAffinityMapOptions(t *testing.T) {
 	if err != nil {
 		t.Errorf("failed to marshal csi config info %v", err)
 	}
-	tmpConfPath := util.CsiConfigFile
-	err = os.Mkdir("/etc/ceph-csi-config", 0o600)
-	if err != nil {
-		t.Errorf("failed to create directory %s: %v", "/etc/ceph-csi-config", err)
-	}
+	tmpConfPath := t.TempDir() + "/ceph-csi.json"
+
 	err = os.WriteFile(tmpConfPath, csiConfigFileContent, 0o600)
 	if err != nil {
 		t.Errorf("failed to write %s file content: %v", util.CsiConfigFile, err)

